src/main/java/io/stargate/sgv2/jsonapi/service/operation/query/ColumnAssignment.java

```diff
@@ -45,12 +59,18 @@ public class ColumnAssignment implements CQLAssignment {
    * Create a new instance of the class to set the {@code column} to the {@code value} in the
    * specified {@code tableMetadata}.
    *
+   * @param updateToAssignment The BiFunction to add update to the ongoing assignment.
    * @param tableMetadata The {@link TableMetadata} for the target table.
    * @param column The name of the column to set.
    * @param value the {@link JsonLiteral} value created by shredding the value from the update
    *     clause in the request.
    */
-  public ColumnAssignment(TableMetadata tableMetadata, CqlIdentifier column, JsonLiteral<?> value) {
+  public ColumnAssignment(
+      BiFunction<OngoingAssignment, CqlIdentifier, UpdateWithAssignments> updateToAssignment,
+      TableMetadata tableMetadata,
+      CqlIdentifier column,
+      JsonLiteral<?> value) {
+    this.updateToAssignment = updateToAssignment;
```

Comment:
I am confused why this one assignment has both the column+value AND the updateToAssignment . 

it looks like the *ToAssignment classes should be subclasses of ColumnAssignment